include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -151,23 +154,311 @@ class alignment_file_format_sam
+        // Alignment object construction
+        // Note that the query sequence in get<1>(align) has already been filled while reading Field 10.
+        if constexpr (!detail::decays_to_ignore_v<align_type>)
+        {
+            if (!cigar.empty() && !std::ranges::empty(get<1>(align))) // only parse alignment if cigar and seq was given
+            {
+                if constexpr (!detail::decays_to_ignore_v<ref_seqs_type>)
+                {
+                    assert(header.ref_dict.count(ref_id_tmp) != 0); // taken care of in check_and_assign_ref_id()
+
+                    size_t pos = header.ref_dict[ref_id_tmp]; // get index for reference sequence
+
+                    assert(static_cast<size_t>(ref_offset_tmp + ref_length) <= std::ranges::size(ref_seqs[pos]));
+
+                    // copy over unaligned reference sequence part
+                    assign_unaligned(get<0>(align),
+                                     ref_seqs[pos] | view::slice(ref_offset_tmp, ref_offset_tmp + ref_length));
+                }
+                else
+                {
+                    using unaligned_t = remove_cvref_t<detail::unaligned_seq_t<decltype(get<0>(align))>>;
+                    auto dummy_seq    = ranges::view::repeat_n(value_type_t<unaligned_t>{}, ref_length) |
+                                                               std::view::transform(detail::access_restrictor_fn{});
+                    static_assert(std::Same<unaligned_t, decltype(dummy_seq)>,
+                                  "No reference information was given so the type of the first alignment tuple position"
+                                  "must have an unaligned sequence type of a dummy sequence ("
+                                  "ranges::view::repeat_n(dna5{}, size_t{}) | "
+                                  "std::view::transform(detail::access_restrictor_fn{}))");
+
+                    assign_unaligned(get<0>(align), dummy_seq); // assign dummy sequence
+                }
+
+                // insert gaps according to the cigar information
+                detail::alignment_from_cigar(align, cigar);
+            }
+            else // not enough information for an alignment, assign an empty view/dummy_sequence
+            {
+                if constexpr (!detail::decays_to_ignore_v<ref_seqs_type>) // reference info given
+                {
+                    assert(std::ranges::size(ref_seqs) > 0); // we assume that the given ref info is not empty
+                    assign_unaligned(get<0>(align), ref_seqs[0] | view::slice(0, 0));
+                }
+                else
+                {
+                    using unaligned_t = remove_cvref_t<detail::unaligned_seq_t<decltype(get<0>(align))>>;
+                    assign_unaligned(get<0>(align), ranges::view::repeat_n(value_type_t<unaligned_t>{}, 0)
+                                                    | std::view::transform(detail::access_restrictor_fn{}));
```

Comment:
why is it repeated 0 times and not infinite?

Reply:
because I want a view of size 0 a.k.a an empty view and not an infinite view? o.O